addons/mail/models/mail_thread.py

```diff
@@ -1869,7 +1870,7 @@ def _message_post_process_attachments(self, attachments, attachment_ids, message
                 # keep cid and name list synced with attachement_values_list length to match ids latter
                 cid_list.append(cid)
                 name_list.append(name)
-            new_attachments = self.env['ir.attachment'].create(attachement_values_list)
+            new_attachments = self.env['ir.attachment'].sudo().create(attachement_values_list)
```

Comment:
@mart-e we are writing a test to cover the addition of a 'sudo' when updating the "main attachment" of a record, when posting a message (see '_message_set_main_attachment_id'). While looking at the code processing attachments, we saw that part of attachments management is already done in sudo (see the 'filtered' here above that generated a subset in sudo, used to update their model / res_id). Those sudo come notably to the chatter-specific rules, allowing you to post when you can read on some models for example or when multicompany is involved.

Shouldn't we also sudo this create here, to be coherent ? We checked and normally this method is called in flows where (at least read) access on the related record is already done. But as playing with attachments often leads to some issues, I ping you to have your thoughts on this.

If you don't understand what I mean, that's because I am not well reveilled.

Bisous !